distributed/worker.py

```diff
@@ -452,7 +452,7 @@ def __init__(
         # Target interface on which we contact the scheduler by default
         # TODO: it is unfortunate that we special-case inproc here
         if not host and not interface and not scheduler_addr.startswith("inproc://"):
-            host = get_ip(get_address_host(scheduler_addr))
+            host = get_ip(get_address_host(scheduler_addr.split("://")[-1]))
```

Comment:
Why was this change needed?

Reply:
It's not entirely related.  I was testing this with another workload and ran into a small bug here.